core/src/main/java/org/mule/runtime/core/exception/TemplateOnErrorHandler.java

```diff
@@ -179,6 +181,9 @@ protected void doInitialise(MuleContext muleContext) throws InitialisationExcept
         new DefaultMessageProcessorChainBuilder(this.flowConstruct);
     try {
       configuredMessageProcessors = defaultMessageProcessorChainBuilder.chain(getMessageProcessors()).build();
+      if (configuredMessageProcessors instanceof FlowConstructAware) {
```

Comment:
It's be good to add a utility method to LifecycleUtils for this, so we are not repeating it everywhere.